src/template_finder.py

```diff
@@ -60,10 +60,37 @@ def stored_templates() -> dict[Template]:
             )
     return templates
 
+@cache
+def stored_templates_by_dir() -> dict[Template]:
```

Comment:
I dont like that we bascially copy the stored_templates() functionality here. I'd rather go for adding the dir normally to TEMPLATE_PATHS and to get all the specific template names loop through folder and get names of the files in a list.

Reply:
Then we also can ditch the search_all_templates() function.

Reply:
stored_templates_by_dir and search_all_templates have been removed

Reply:
@mgleed I added some asserts but they're very basic; please validate